### PR TITLE
リファクタリング

### PR DESCRIPTION
**Pull request by imaizume**
概要
---

このPR(Pull Request)ではコードのリファクタリングを行い、特定の画面幅で現れる横スクロール関連のバグを修正します。
作業が終わるまで、下の**マージボタンは絶対に押さないで下さい**

@dptadfighter 君へのアドバイスも書き込んでいくので暇があったら見てください。

### CODE REVIEW

**Comment by imaizume** [comment]
作業方針
---

1. 全過程でインデントを正しく直し、とじタグが誤った箇所も直していく
1. ヘッダーナビゲーションが正しいかをチェック
1. `container`クラスの使い方が違うので修正
1. ヘッダーと`<video>`で読み込んでいるリソース群のパスを修正 AND `</body>` 直前へリプレイス
1. 現在`container-fluid`指定されている部分を正しく解体していく
1. `<svg>`や`<path>`などの描画処理をスクリプトへ移せないか検討する

**Comment by imaizume** [comment]
@dptadfighter まず、全体的にインデント(コード左にTABであけるスペース)がバラバラです。
インデントはHTMLではタグの始まりと終わりを表す重要な目印になるので、初心者のうちは特に
もしDreamweaverを使っているのであればソースフォーマットの機能を利用しましょう。
同様の機能は他の開発用エディタ(IDE)にも通常デフォルトでセットされています

https://blog.sus-happy.net/201006/dreamweaver-tips/

**Comment by imaizume** [comment]
```html
<div>
<p>hello world</p>
<ul>
    <li>apple</li>
<li>banana</li>
  <ul>
</div>
```

よりも

```html
<div>
  <p>hello world</p>
  <ul>
    <li>apple</li>
    <li>banana</li>
  <ul>
</div>
```

が見やすい

**Comment by gamerinshaft** [comment]
:+1:

**Comment by imaizume** [comment]
@dptadfighter また別件になりますが、初心者にはHTMLの基礎を理解する上であまりDreamweaverはおすすめしません。
可能な限り直打ち(ORタグ名のみの補完くらいができる)エディタを使いましょう。
そのほうがタグ名やプロパティ名を覚えるし、打っている過程でインデントや閉じタグのミスにも気付きやすくです。

* [Sublime Text](http://www.sublimetext.com/3)
* [Terapad](http://www.forest.impress.co.jp/library/software/terapad/)
* [Notepad++](http://www.forest.impress.co.jp/library/software/notepadplus/)

File: index.html [hunk 1]
```diff
@@ -83,11 +84,6 @@
         </div>
         <!-- Collect the nav links, forms, and other content for toggling -->
         <div class="collapse navbar-collapse" id="inverseNavbar1">
-          <ul class="nav navbar-nav">
```
Comment (imaizume) [inline index.html]: 不要なナビゲーションメニューがあったので削除しました。何もアイテムが無いとプルダウンメニューで無駄な横線が入ってしまします。

File: index.html [hunk 2]
```diff
@@ -23,7 +23,6 @@
   <![endif]-->
 
     <!-- jQuery.jsの読み込み -->
-    <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.3.2/jquery.min.js"></script>
```
Comment (incheon) [inline index.html]: @dptadfighter フッター付近でより新しいjQueryを読み込んでいるので、こちらは不要です。

**Comment by incheon** [comment]
@dptadfighter 直前の2つのコミットについて解説します。

コミットID https://github.com/unitus-jp/static-site/pull/3/commits/4144a45e05affcfdbf14d8f15e87209502e433d5
---

`<h4>`を使っていましたが、これはこの場合は適切ではありません。理由は**h1~h6が見出しを意味する要素**だからです。この文章はWebページ的には見出しではなく、本文の一部と考えられます。

もし見た目を`<h4>`のように少し大きめかつ太めにしたいのであれば、[Bootstarapには`.h1` ~ `.h6`
というクラスがあります](http://getbootstrap.com/css/#type)ので、こちらで見た目を指定するようにしましょう。

また改行用のHTMLタグは`<br>`になりますので、極力改行目的で実体参照の`&nbsp;`は使わないようにしましょう。

http://digilog.client.jp/html/br-space.html

コミットID https://github.com/unitus-jp/static-site/pull/3/commits/025d9972d985eea198f821f1d46000392b7c20e6
---

h1~h6の中でも、最も大きな見出しの`<h1>`には**同じページで1度しか出現してはいけない**という特殊なルールがあります。つまるところ、`<title>`と同じような役割だと思ってもらえば良いでしょう。よって複数箇所の出現は避けるようにします。

http://whitehatseo.jp/%E8%A6%8B%E5%87%BA%E3%81%97%E3%82%BF%E3%82%B0-hn%E3%82%BF%E3%82%B0-h1%EF%BD%9Eh6%E3%81%AE%E4%BD%BF%E3%81%84%E6%96%B9/

**Comment by incheon** [comment]
@dptadfighter 

コミットID https://github.com/unitus-jp/static-site/pull/3/commits/1146bfa4408a65d71538d8e6dd89e653630b3484
---


SVGを使ってロゴを書いているとは、大変素晴らしいです、GOODJOB! :+1: 
僕もSVG自体についてはあまり勉強していなかったので、これをきっかけに調べて見ようと思います。どうもありがとう :bow: 

この素敵なロゴを、もっとメンテナンスしやすく、かつindex.htmlを見やすくするために、別ファイル(images/logo.svg)に切り分けて、``タグで読込させています。

**Comment by imaizume** [comment]
上の @incheon なるものは自分の古い別垢です、すみません :droplet:

File: index.html [hunk 3]
```diff
@@ -95,14 +95,11 @@ <h1 class="text-center overview-title"><span class="t2">「</span>UNITUSは環
...
       <div class="col-sm-1"></div>
       <div class="col-sm-10">
-        <h4 class="text-left">UNITUSは環境です。</h4>
-        <h4 class="text-left">私達は自主性を育みます。</h4>
-        <h4 class="text-left">&nbsp;</h4>
-        <h4 class="text-left">入るのも自由、去るのも自由、やるもやらないもあなたの自由です。</h4>
-        <h4 class="text-left">本当の楽しみは、与えられるものではないです。自分で考え、勝ち取るものです。</h4>
-        <p>&nbsp;</p>
-        <h4 class="text-left">自分がやりたいことに仲間と共に挑戦してください。</h4>
-        <br><br>
```
Comment (imaizume) [inline index.html]: @dptadfighter `<br>`を消したことによって、この行の下にあったスペースがなくなっていますが、後ほどgridを整理する段階でCSSを使って復活させますので安心して下さい。多くの場合、行間を確保するにはHTMLタグを利用するよりもCSS(`margin`, `padding`, `line-height`)の指定が適切です。

File: index.html [hunk 4]
```diff
@@ -76,14 +76,12 @@
   </div>
   <div class="col-sm-6 row-center-content">
     <div class="row row-margin">
-      <div class="col-sm-1"></div>
-      <div class="col-sm-10">
+      <div class="col-sm-10 col-sm-offset-1">
```
Comment (imaizume) [inline index.html]: @dptadfighter bootstrapには`col-sm-offset-1`のような、**Gridで左右に余白列を持たせるためのクラス**が[デフォルトで定義されている](http://getbootstrap.com/css/#grid-offsetting)ので、空の列を作るよりもこちらを使うほうが自然です。

**Comment by imaizume** [comment]
@taikit とりあえずここで一度作業止めます。作業方針に示した要件は全て満たしたはずなので、ローカルにcheckoutして確認してみてください。

@dptadfighter コミットが多いので、あとでコメントするか直接口頭で解説します。

File: index.html [hunk 5]
```diff
@@ -107,7 +107,11 @@ <h2 class="h1 text-center overview-title"><span class="t2">「</span>UNITUSは
   <div id="particles-js"></div>
   <div class="row">
     <div class="col-sm-6">
-      <img src="images/switch-logo.png" alt="Placeholder image" width="100" height="100" class="img-responsive switch-logo"><h2 class="text-right product-txt">Switch</h2>
+      <div>
+        <img src="images/switch-logo.png" alt="Placeholder image"
+             class="img-responsive switch-logo" style="max-width: 75px;display: inline-block;vertical-align: middle;" />
```
Comment (imaizume) [inline index.html]: 画像とテキストを共に`display-inline`で高さと幅を持たせて`vertical-align: middle`を指定する。それらを`<div>`で囲むと上下中央配置になって見た目が少し綺麗になります。

File: index.html [hunk 6]
```diff
@@ -136,12 +136,15 @@ <h2 class="h1 contact-title" style="line-height: 2em;">自らのスキルを活
       <div class="row contact-switch">
         <div class="col-xs-6">
           <form action="mailto:unitus.tus@gmail.com?subject=" method="post">
-            <button type="submit" class="btn btn-lg btn-primary center-block mail-box button-shadow"><span class="glyphicon glyphicon-envelope" aria-hidden="true"></span> Mail<br>
+            <button type="submit" class="btn btn-lg btn-primary center-block mail-box button-shadow">
+              <span class="glyphicon glyphicon-envelope" aria-hidden="true"></span><span style="margin-left: .5em;">Mail</span>
```
Comment (imaizume) [inline index.html]: @dptadfighter グリフィコンやアイコンと文字の間のスペースは文字(" ")で与えると見難いですし、改行等した場合に反映されなくなるなどの影響が出やすいので、テキストかアイコンのどちらかの左右の余白を指定してあげます。

File: stylesheet.css [hunk 7]
```diff
@@ -9,9 +9,9 @@
 #video-background {
 /*  動画を背景に設定するスタイルシート  */
   position: fixed;
-  right: 0;
+  left: -50%;
```
Comment (imaizume) [inline stylesheet.css]: 動画を画面サイズの50%左外側にずらします

File: stylesheet.css [hunk 8]
```diff
@@ -9,9 +9,9 @@
...
   bottom: 0;
-  min-width: 100%;
+  min-width: 150%;
```
Comment (imaizume) [inline stylesheet.css]: ずらした分の50%分を拡大して表示します。
この`postion`や`margin`のマイナス指定は時々出てくるテクニックなので覚えてくと便利でしょう。

File: index.html [hunk 9]
```diff
@@ -59,16 +59,18 @@
     </nav>
   </div>
 
-  <video class="hidden-xs" id="video-background"
-         autoplay loop poster="images/background picture for smartphone.png" muted>
-    <source src="http://unitus-static.eu-gb.mybluemix.net/timelapse.webm" type="video/webm" />
-    <source src="http://unitus-static.eu-gb.mybluemix.net/timelapse.mp4" type="video/mp4" />
-  </video>
   <div class="container-fluid top">
+    <video class="" id="video-background"
```
Comment (imaizume) [inline index.html]: @dptadfighter `hidden-xs`を誤って消していました、すみません :bow: 
あとで復活させておきます。

File: stylesheet.css [hunk 10]
```diff
@@ -220,7 +220,6 @@ margin-top: 62px;
 	}
 
 #map {
-	width: 100vw;
```
Comment (imaizume) [inline stylesheet.css]: @dptadfighter `vw`という単位は、viewport(画面に表示されるまえに一度作られる仮想的な画面サイズのこと)の1%を表す長さの単位ですが、あまり見かけたたことがなく、かつ自分が普段あまりviewport触れないので、今度調べて改めて話します。

File: index.html [hunk 11]
```diff
@@ -71,10 +71,10 @@
 
   <div class="container-fluid overview card" id="principle">
     <div class="row overview-row">
-      <div class="col-sm-6 row-center-title">
```
Comment (imaizume) [inline index.html]: 特にCSSで指定がなかったので、不要なクラスは削除しました。

File: index.html [hunk 12]
```diff
@@ -100,8 +101,12 @@ <h2 class="h1 text-center overview-title"><span class="t2">「</span>UNITUSは
           </div>
         </div>
       </div>
+      <div class="col-sm-6 hidden-xs">
```
Comment (imaizume) [inline index.html]: @dptadfighter  bootstrapでのテクニックですが、このようにxs以下でのみ表示するところと、`sm`以上で表示する部分を切り分ければ、レイアウトを大きく変更可能です。口頭で説明するのが早いので、もしコードを読んでわからなければ質問して下さい。

File: index.html [hunk 13]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html lang="jp">
+<html lang="ja">
```
Comment (imaizume) [inline index.html]: @dptadfighter ここでは言語の記号を指定ます。言語とは`en` (English)や`ja` (Japanese)などの略称で、これと別にロケール(地域)を表す記号も有ります。例えば`us`(United States)や `jp` (Japan)などです。いろんな言語で出てくるので、使っているとなれると思います。とりあえずhtml5での言語属性は`lang="en"`又は`lang="en"`と覚えておきましょう。

File: index.html [hunk 14]
```diff
@@ -152,6 +151,7 @@ <h2 class="h3 map-adress" style="margin-top: 0;">ADRESS</h2>
   </div>
   </div>
 
+  <script src="http://maps.google.com/maps/api/js?sensor=true"></script>
```
Comment (imaizume) [inline index.html]: `<script>`タグ(JavaScript)は最後、`<link>`(スタイルシート)や`<meta>`は頭とおぼえておきましょう。理由は後日説明します。

**Comment by imaizume** [comment]
@gamerinshaft と話をして、このPRではスタイルシートの整理までしてからマージして渡すことにしました。

**Comment by imaizume** [comment]
@taikit @gamerinshaft @dptadfighter とりあえずここで作業を終わりにします :pencil2: まだもう少しだけリファクタリングの余地があるけど、セレクタの名前とかのレベルなので。一度手元に落としてもらって確認して問題ないようならマージして下さい :octocat:

**Comment by imaizume** [comment]
ライブラリの読み込みをCDNにしたのもありますが、それを差し引いても削除の方が多いですね :scissors:   リファクタリングあるあるです。

diff

File: css/stylesheet.css [hunk 15]
```diff
@@ -125,7 +125,6 @@
 
 .switch {
     z-index: 0;
-    display: relative;
```
Comment (imaizume) [inline css/stylesheet.css]: 多分単純なミスだと思いますが、エディタの補完機能を使えばこうしたミスは少なくなります。

**Comment by gamerinshaft** [comment]
svgって画像に出来ないかな？
コードが読みづらくなっちゃうのちょっといや😞

**Comment by dptadfighter** [comment]
お疲れ様でした。じっくりと見直します(･･;)

**Comment by dptadfighter** [comment]
たしか今泉先輩がsvgの画像に差し替えてくれました。

**Comment by gamerinshaft** [comment]
オシャレなサイトまとめとかみて、それを自分で作ってみるとかするといい勉強になるよ！

**Comment by gamerinshaft** [comment]
そうなんだ！ありがとう！

**Comment by dptadfighter** [comment]
なるほど！ミス慶應2015のコードを見てみます

**Comment by gamerinshaft** [comment]
あれとかは特殊な例だから、もっと落ち着いてるサイトがいいよ。
ミス慶応は画像が映えるからああゆうレイアウトができるわけで😎

**Comment by imaizume** [comment]
@gamerinshaft @dptadfighter 大丈夫だと思うけど補足すると、SVGは別ファイルに抽出して``で読み込みするようにしただけで、画像にしたわけではないです。まぁPNG変換した方が扱いやすいのならそうしても良いかもね。

**Comment by gamerinshaft** [comment]
おっけー、ありがとう！
